python/ray/_private/services.py

```diff
@@ -1889,17 +1893,35 @@ def start_ray_client_server(redis_address,
             no redirection should happen, then this should be None.
         redis_password (str): The password of the redis server.
         server_type (str): Whether to start the proxy version of Ray Client.
+        serialized_runtime_env (str|None): If specified, the serialized
+            runtime_env to start the client server in.
 
     Returns:
         ProcessInfo for the process that was started.
     """
+    root_ray_dir = Path(__file__).resolve().parents[1]
+    setup_worker_path = os.path.join(root_ray_dir, "workers",
+                                     ray_constants.SETUP_WORKER_FILENAME)
+    conda_shim_flag = (
+        "--worker-setup-hook=" + ray_constants.DEFAULT_WORKER_SETUP_HOOK)
+
     command = [
-        sys.executable, "-m", "ray.util.client.server",
+        sys.executable,
+        setup_worker_path,
+        conda_shim_flag,  # These two args are to use the shim process.
+        "-m",
+        "ray.util.client.server",
         "--redis-address=" + str(redis_address),
-        "--port=" + str(ray_client_server_port), "--mode=" + server_type
+        "--port=" + str(ray_client_server_port),
+        "--mode=" + server_type
     ]
     if redis_password:
         command.append("--redis-password=" + redis_password)
+
+    if serialized_runtime_env:
+        command.append("--serialized-runtime-env=" + serialized_runtime_env)
+    if session_dir:
+        command.append(f"--session-dir={session_dir}")
```

Comment:
I get errors due to this. It looks like client_server doesn't support session_dir